src/mission_companion.cpp

```diff
@@ -476,11 +476,35 @@ bool talk_function::display_and_choose_opts(
         }
 
         int name_offset = 0;
-        calcStartPos( name_offset, sel, info_height, folded_names_lines );
+        size_t sel_pos = 0;
+        // Translate from entry index to line index
+        for (size_t i = 0; i < sel; i++) {
+            sel_pos += folded_names[i].size();
+        }
+
+        calcStartPos( name_offset, sel_pos, info_height, folded_names_lines );
+
+        int name_index = 0;
+        // Are we so far down the list that we bump into the end?
+        bool last_section = folded_names_lines < info_height || folded_names_lines - info_height <= size_t(name_offset);
+
+        // Translate back from desired line index to the corresponding entry, making sure to round up near the end to ensure the last line gets included.
```

Comment:
Since astyle doesn't do it, please manually break comment lines at 100 characters.  Thank you.,